### PR TITLE
implemented CRUD operations for department options

### PR DESCRIPTION
**Pull request by ChimezieGlory**

### CODE REVIEW

File: BusinessLayer/Services/DepartmentOptionService.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            _configuration = configuration;
+            baseUrl = configuration.GetValue<string>("Url:root");
+        }
+
+        public async Task<ResponseModel> AddDepartmentOption(DepartmentOptionDto model)
+        {
+            try
+            {
+                DepartmentOption deptOption = new DepartmentOption();
+                //var dpo_slug = Utility.GenerateSlug(model.Name);
```
Comment (GoodspeedMiracle) [inline BusinessLayer/Services/DepartmentOptionService.cs:31]: remove commented code

File: BusinessLayer/Services/DepartmentOptionService.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                var doesExist = await _context.DEPARTMENT_OPTION.Where(dpo => dpo.Id == model.Id).FirstOrDefaultAsync();
+
+               // var doesExist = await _context.DEPARTMENT_OPTION.Where(f => f.Name == dpo_slug).FirstOrDefaultAsync();
```
Comment (GoodspeedMiracle) [inline BusinessLayer/Services/DepartmentOptionService.cs:34]: remove commented code

File: BusinessLayer/Services/DepartmentOptionService.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            catch (Exception ex)
+            {
+
+                throw ex;
+            }
+        }
+
+        public async Task<IEnumerable<DepartmentOptionDto>> GetDepartmentOptionsByDepartmentId(long DepartmentId, bool isAdmin)
+        {
+            if (isAdmin)
```
Comment (GoodspeedMiracle) [inline BusinessLayer/Services/DepartmentOptionService.cs:111]: The if condition here is unnecessary. Just return the department options

File: BusinessLayer/Services/DepartmentOptionService.cs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                return response;
+            }
+            catch (Exception ex)
+            {
+
+                throw ex;
+            }
+        }
+
+        public async Task<IEnumerable<DepartmentOptionDto>> GetDepartmentOptionsByDepartmentId(long DepartmentId, bool isAdmin)
```
Comment (GoodspeedMiracle) [inline BusinessLayer/Services/DepartmentOptionService.cs:109]: remove the isAdmin parameter here. Not neccessary

**Review by MiracleGodspeed — CHANGES_REQUESTED** [review]
see commnets and implement changes

**Pull request merged**